core/src/test/java/org/apache/calcite/test/TopDownOptTest.java

```diff
@@ -693,6 +678,37 @@
         .removeRule(EnumerableRules.ENUMERABLE_JOIN_RULE)
         .check();
   }
+
+  // push sort to left input
+  @Test void testBatchNestedLoopJoinLeftOuterJoinPushDownSort() {
```

Comment:
Not adding too many tests for EnumerableBatchNestedLoopJoin because it reuses EnumerableNestedLoopJoin's code and those code was tested.

Add two test only to verify that trait propagation in EnumerableBatchNestedLoopJoin is working.